### PR TITLE
isl 0.24

### PR DESCRIPTION
**Pull request by paperchalice**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by paperchalice** [comment]
`flann` official site is down.

**Comment by carlocab** [comment]
You should probably drop the `libbi` commit, as I'm not sure we want to rebuild that against the new Big Sur SDK yet. (It'll also need rebuilding once the Intel Big Sur runner has been updated to 11.3.)

The `mpich` issue will also be fixed by #73062, so you can just drop that commit and rebase once #73062 is merged.

**Comment by carlocab** [comment]
Please remove the `perl` revision bump for `libbi`.

File: Formula/isl.rb
Comment (carlocab) [inline Formula/isl.rb:41]:
```suggestion
    site_packages = Language::Python.site_packages("python3")
```
This works too, and makes the formula easier to update for `python@3.10`.

File: Formula/flann.rb
Comment (carlocab) [inline Formula/flann.rb:53]: Would be good to try to keep the old test. We can use the new one you propose for now just to check that `flann` isn't broken by the `isl` version bump, though.

File: Formula/isl.rb
Comment (paperchalice) [reply Formula/isl.rb:51]: Some C++ interface like `cpp-checked.h` are not installed by the `make install`, need to install them manually.

File: Formula/isl.rb
Comment (paperchalice) [reply Formula/isl.rb:70]: Just test the C++ interface

File: Formula/flann.rb
Comment (paperchalice) [reply Formula/flann.rb:53]: Currently resource URL are unreachable, ERROR 500.

Comment (carlocab) [reply Formula/flann.rb:53]: Yes, saw your comment: https://github.com/Homebrew/homebrew-core/pull/76444#issuecomment-830805188

File: Formula/isl.rb
Comment (paperchalice) [reply Formula/isl.rb:34]: It needs some `clang` libraries to build binding generator.

**Comment by paperchalice** [comment]
```
==> Testing libbi
/usr/bin/sandbox-exec -f /private/tmp/homebrew20210503-79670-2gsqxd.sb ruby -W1 -- /opt/homebrew/Library/Homebrew/test.rb /opt/homebrew/Library/Taps/homebrew/homebrew-core/Formula/libbi.rb --verbose --retry
==> /opt/homebrew/Cellar/libbi/1.4.5_2/bin/libbi sample @test.conf
Stash.c: loadable library and perl binaries are mismatched (got handshake key 0xc800080, needed 0xc700080)
==> Testing libbi (again)
/usr/bin/sandbox-exec -f /private/tmp/homebrew20210503-79917-pm6fly.sb ruby -W1 -- /opt/homebrew/Library/Homebrew/test.rb /opt/homebrew/Library/Taps/homebrew/homebrew-core/Formula/libbi.rb --verbose --retry
==> /opt/homebrew/Cellar/libbi/1.4.5_2/bin/libbi sample @test.conf
Stash.c: loadable library and perl binaries are mismatched (got handshake key 0xc800080, needed 0xc700080)
Error: libbi: failed
```

**Comment by carlocab** [comment]
That's expected. https://github.com/Homebrew/brew/issues/11275

Also seen in a number of other PRs.

File: Formula/isl.rb
Comment (fxcoudert) [reply Formula/isl.rb:51]: This need to be reported upstream, so it gets fixed by them and we can remove this ugliness in the next version.

File: Formula/isl.rb
Comment (fxcoudert) [reply Formula/isl.rb:34]: Let's build without those, we don't want to depend on llvm itself

**Comment by carlocab** [comment]
This should be good to merge when you drop the `openmodelica` commit and add a comment linking to an upstream report for the manual header installation.

**Comment by paperchalice** [comment]
Reported to mailing list (`isl` and `pet` use the same mailing list)

**Comment by carlocab** [comment]
> Reported to mailing list (`isl` and `pet` use the same mailing list) the only reply is
> 
> ```
> Sven Verdoolaege <sven.verdoolaege.isl@gmail.com>
> 
> I guess it's just that nobody ever requested them to be installed.
> I don't use them myself.
> 
> skimo
> ```

This makes it sound like these headers aren't useful then. What are the extra headers used for?

**Comment by paperchalice** [comment]
These headers are contributed by `llvm`, the component `polly` use the C++ interface.

**Comment by carlocab** [comment]
It might help if you contributed a patch upstream to install these headers as well with `make install`.

**Review by carlocab — COMMENTED** [review]
You should also drop the `openmodelica` commit, since 1.17.0 does not build.

**Comment by paperchalice** [comment]
Drop the C++ headers now. In fact python binding doesn't need `python@3.9`, it is distributed by the source code, but is still not installed by `make install`. About the patch I don't think I can write it because I'm not familiar to automake.😰

`openmodelica` needs many upstream commits to build.

**Review by carlocab — APPROVED** [review]
Thanks @paperchalice.

**Pull request closed without merging**
